lib/Jsrt/Jsrt.cpp

```diff
@@ -1422,11 +1422,8 @@ CHAKRA_API JsPreventExtension(_In_ JsValueRef object)
 }
 
 CHAKRA_API JsHasOwnPropertyCommon(Js::ScriptContext * scriptContext, _In_ JsValueRef object,
-    _In_ const Js::PropertyRecord * propertyRecord, _Out_ bool *hasOwnProperty,
-    TTDRecorder& _actionEntryPopper)
+    _In_ const Js::PropertyRecord * propertyRecord, _Out_ bool *hasOwnProperty)
 {
-    PERFORM_JSRT_TTD_RECORD_ACTION(scriptContext, RecordJsRTHasOwnProperty, propertyRecord, object);
```

Comment:
what was broken w/ having this here?

Reply:
concern here is that every caller of `JsHasOwnPropertyCommon` will now have to "do the right thing" for TTD, namely either assert of record.  Is there some way to have the right logic consolidated in `JsHasOwnPropertyCommaon` , (record if we can, assert if not), independent of caller?

Reply:
Any instances of VALIDATE_INCOMING_OBJECT can also marshal the object to the current context. The record actions need to be the value prior any of these calls to capture the original pointer.

Reply:
Thanks Kyle, a few questions:

1.  What problem does this fix today?  Is something broken?
2.  With this change, what are the implications on someone trying to use TTD?  
3.  Is it possible to capture the value before `VALIDATE_INCOMING_OBJECT`, and include that as a param into the `*Common*` methods?   This way, the right value will get recorded, and the API signature provides some assurance that any future callers will "do the right thing".

LMK if I'm mis-understanding something here.  Happy to talk in on phone or in office.

Reply:
Attempted to update the commit description again:

> Any instances of VALIDATE_INCOMING_OBJECT can also marshal the object
> to the current context. The record actions need to be the value prior
> any of these calls to capture the original pointer.
>
> These methods were recently added and in the process regressed the
> existing methods. Partially backing out the change so we can properly
> address the TTD implementation for these methods.

Reply:
OK, is there a bug opened to track getting this fixed?  Do any existing tests fail with these asserts now?

Reply:
Answers to your questions:
1. These were causing `vm` module scenarios to fail in TTD, it regressed from v1.7.1 which was the last supported build with TTD.
2. It causes the record to fail if those JsObject* functions are called from a native module, they are basically just fast-path implementations and introduced later in the v1.7.* cycle, so they aren't currently used in node-chakracore.
3. The real fix is not that simple, the record will need to be different for the JsObject* calls because they need to capture the JsValueRef during record and generate a JsPropertyIdRef during replay.  This also requires that we introduce some new logic during replay so it's not a trivial fix.

Reply:
clarified this offline with @kfarnung.  This change fixes a regression in TTD that was causing some internal tests to fail.  The new "assert" code paths aren't hit in these tests.